src/protocol/properties.rs

```diff
@@ -81,8 +82,29 @@ pub struct Location {
     pub range: Range,
 }
 
-#[derive(Serialize, Deserialize, Clone, Default)]
+#[derive(Serialize_repr, Deserialize_repr, PartialEq, Clone)]
+#[repr(u8)]
+pub enum TextDocumentSyncKind {
+    /**
+     * Documents should not be synced at all.
+     */
+    None = 0,
+    /**
+     * Documents are synced by always sending the full content of the document.
+     */
+    Full = 1,
+    /**
+     * Documents are synced by sending the full content on open. After that only incremental
+     * updates to the document are sent.
+     */
+    Incremental = 2,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
 pub struct ServerCapabilities {
+    #[serde(rename = "textDocumentSync")]
+    pub text_document_sync: TextDocumentSyncKind,
```

Comment:
Maybe I'm misunderstanding the spec, but it looks like this is meant to just specify one of the three kinds of sync, not the whole serialized enum, right?

Reply:
Nevermind, you're right, lol.

Reply:
Per the LSP spec, it defaults to `None` if not set, so changing it to `Incremental` should be more correct now that we're syncing file state between the server.

Reply:
for the time lets set this to full and make a card to track switching it